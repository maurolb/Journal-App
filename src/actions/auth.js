import Swal from 'sweetalert2';
import { getAuth, updateProfile ,signInWithPopup, createUserWithEmailAndPassword, signInWithEmailAndPassword } from 'firebase/auth';
import { googleProvider } from '../firebase/firebase-config';
import { types } from "../types/types";
import { finishLoading, startLoading } from './ui';
import { noteLogout } from './notes';

export const startLoginEmailPassword = ( email, password ) => {
  return ( dispatch ) => {

    dispatch( startLoading() );

    const auth = getAuth();
    return signInWithEmailAndPassword( auth, email, password )
      .then( ({ user }) => {
        dispatch( login( user.uid, user.displayName ) );
        dispatch( finishLoading() );
      })
      .catch(( err ) => {
        console.log( err );
        dispatch( finishLoading() );
        Swal.fire('Error', err.message, 'error');
      });
  };
};

export const startRegisterWithEmailPasswordName = ( email, password, name ) => {
  return ( dispatch ) => {
    const auth = getAuth();
    createUserWithEmailAndPassword( auth, email, password )
      .then( async ({ user }) => {
        await updateProfile( user, {displayName: name} );
        dispatch( login( user.uid, user.displayName ) );
      })
      .catch(( err ) => {
        console.log( err );
        Swal.fire('Error', err.message, 'error');
      });
      
  };
};

export const startGoogleLogin = () => {
  return ( dispatch ) => {
    const auth = getAuth();
    signInWithPopup( auth, googleProvider )
      .then(({ user }) => {
          dispatch( login( user.uid, user.displayName ) );
      });
  };
};

export const login = ( uid, displayName ) => {
  return {
    type: types.login,
    payload: {
      uid,
      displayName
    }
  };
};

export const startLogout = () => {
  return async ( dispatch ) => {
    await getAuth().signOut();
    dispatch( logout() );
    dispatch( noteLogout() );
  };
};

export const logout = () => ({
  type: types.logout
});