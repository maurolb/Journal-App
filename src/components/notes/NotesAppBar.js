import moment from 'moment';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { startSaveNote, startUploading } from '../../actions/notes';

export const NotesAppBar = () => {

  const dispatch = useDispatch();
  const { active } = useSelector( state => state.notes );
  const noteDate = moment( active.date );

  const handleSave = () => {
    dispatch( startSaveNote( active ) );
  };

  const handlePictureClick = () => {
    document.querySelector('#fileSelector').click();
  };
  
  const handleFileChange = (e) => {
    const file = e.target.files[0];
    if( file ) {
      dispatch( startUploading( file ) );
    };
  };

  return (
    <div className="notes__appbar">
      <span>{ noteDate.format('MMMM Do YYYY') }</span>

      <input 
        id="fileSelector"
        type="file"
        name="file"
        style={{display: 'none'}}
        onChange={ handleFileChange }
      />

      <div>
        <button 
          className="btn"
          onClick={ handlePictureClick }
        >
          <i className="fas fa-camera"></i> Picture
        </button>

        <button 
          className="btn"
          onClick={ handleSave }
        >
          <i className="fas fa-camera"></i> Save
        </button>
      </div>
    </div>
  );
};
