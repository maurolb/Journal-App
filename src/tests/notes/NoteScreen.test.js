import configureStore from 'redux-mock-store';
import thunk from "redux-thunk"; 
import { mount } from "enzyme";
import { Provider } from "react-redux";
import { NoteScreen } from '../../components/notes/NoteScreen';
import { activeNote } from '../../actions/notes';


jest.mock('../../actions/notes', () => ({
  activeNote: jest.fn()
}));

// jest.mock("../../../actions/notes", () => ({
//   startNewNote: jest.fn()
// }));


const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {
  auth: {
    uid: '1',
    name: 'mauro'
  },
  ui: {
    loading: false,
    msgError: null
  },
  notes: {
    active: {
      id: 1232,
      title: 'tituloTest',
      body: 'bodyTest',
      date: 0
    },
    notes: []
}
} 

let store = mockStore( initialState );
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <NoteScreen />
  </Provider>
);


describe('Tests on NoteScreen component', () => {
  
  test('should show correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });
  

  test('should activeNote', () => {
    wrapper.find('input[name="title"]').simulate('change', {
      target: {
        name: 'title',
        value: 'hola'
      }
    });

    expect( activeNote ).toHaveBeenLastCalledWith(
      1232,
      {
        id: 1232,
        title: 'hola',
        body: 'bodyTest',
        date: 0
      }
    );
  });
  

});
