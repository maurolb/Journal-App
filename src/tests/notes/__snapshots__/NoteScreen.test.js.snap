// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on NoteScreen component should show correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      December 31st 1969
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="some awesome title"
      type="text"
      value="tituloTest"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="what happened today"
      value="bodyTest"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
